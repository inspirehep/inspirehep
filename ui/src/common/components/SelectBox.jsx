import React from 'react';
import { Select } from 'antd';

function SelectBox({ options, virtualScroll = false, ...selectProps }) {
  return (
    <Select dropdownMatchSelectWidth={virtualScroll} showArrow {...selectProps}>
      {options.map((option) => (
        <Select.Option key={option.value} value={option.value}>
          <span
            data-test-id={
              selectProps['data-test-id'] &&
              `${selectProps['data-test-id']}-option-${option.value}`
            }
          >
            {option.display || option.value}
          </span>
        </Select.Option>
      ))}
    </Select>
  );
}

export default SelectBox;
