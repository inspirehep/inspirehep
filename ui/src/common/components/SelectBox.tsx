import React from 'react';
import PropTypes from 'prop-types';
import { Select } from 'antd';
import { SelectOptionsPropType } from '../propTypes';

function SelectBox({ options, virtualScroll = false, ...selectProps }) {
  return (
    <Select dropdownMatchSelectWidth={virtualScroll} showArrow {...selectProps}>
      {options.map(option => (
        <Select.Option key={option.value} value={option.value}>
          <span
            data-test-id={
              selectProps['data-test-id'] &&
              `${selectProps['data-test-id']}-option-${option.value}`
            }
          >
            {option.display || option.value}
          </span>
        </Select.Option>
      ))}
    </Select>
  );
}

SelectBox.propTypes = {
  options: SelectOptionsPropType.isRequired,
  virtualScroll: PropTypes.bool,
};

export default SelectBox;
