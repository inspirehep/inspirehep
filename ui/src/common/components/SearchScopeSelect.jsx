import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SelectBox from './SelectBox';
import { SEARCH_BOX_NAMESPACES, CONFERENCES_NS } from '../../reducers/search';

const SCOPE_OPTIONS = SEARCH_BOX_NAMESPACES.filter(
  value => value !== CONFERENCES_NS
).map(value => ({ value }));

class SearchScopeSelect extends Component {
  render() {
    const { searchScopeName, onSearchScopeChange } = this.props;
    return (
      <SelectBox
        dropdownClassName="header-dropdown"
        onChange={onSearchScopeChange}
        value={searchScopeName}
        options={SCOPE_OPTIONS}
      />
    );
  }
}

SearchScopeSelect.propTypes = {
  onSearchScopeChange: PropTypes.func.isRequired,
  searchScopeName: PropTypes.string.isRequired,
};

export default SearchScopeSelect;
