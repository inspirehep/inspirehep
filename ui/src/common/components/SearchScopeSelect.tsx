import React, { Component } from 'react';
import PropTypes from 'prop-types';

import SelectBox from './SelectBox';
import { SEARCH_BOX_NAMESPACES } from '../../search/constants';

const SCOPE_OPTIONS = SEARCH_BOX_NAMESPACES.map(value => ({ value }));

class SearchScopeSelect extends Component {
  render() {
    const { searchScopeName, onSearchScopeChange } = this.props;
    return (
      <SelectBox
        dropdownClassName="header-dropdown"
        onChange={onSearchScopeChange}
        value={searchScopeName}
        options={SCOPE_OPTIONS}
      />
    );
  }
}

SearchScopeSelect.propTypes = {
  onSearchScopeChange: PropTypes.func.isRequired,
  searchScopeName: PropTypes.string.isRequired,
};

export default SearchScopeSelect;
