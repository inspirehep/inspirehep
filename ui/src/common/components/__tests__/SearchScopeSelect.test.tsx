import React from 'react';
import { shallow } from 'enzyme';

import SearchScopeSelect from '../SearchScopeSelect';
import SelectBox from '../SelectBox';

describe('SearchScopeSelect', () => {
  it('render initial state with all props set', () => {
    const wrapper = shallow(
      <SearchScopeSelect
        onSearchScopeChange={jest.fn()}
        searchScopeName="authors"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onSearchScopeChange on select change', () => {
    const onSearchScopeChange = jest.fn();
    const wrapper = shallow(
      <SearchScopeSelect
        searchScopeName="literature"
        onSearchScopeChange={onSearchScopeChange}
      />
    );
    const onSelectChange = wrapper.find(SelectBox).prop('onChange');
    const newScope = 'authors';
    onSelectChange(newScope);
    expect(onSearchScopeChange).toBeCalledWith(newScope);
  });
});
