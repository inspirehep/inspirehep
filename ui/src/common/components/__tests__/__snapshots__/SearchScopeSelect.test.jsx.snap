// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScopeSelect render initial state with all props set 1`] = `
<SelectBox
  dropdownClassName="header-dropdown"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "value": "literature",
      },
      Object {
        "value": "authors",
      },
      Object {
        "value": "jobs",
      },
    ]
  }
  value="authors"
/>
`;
